src/Stratis.Bitcoin/P2P/PeerAddressManager.cs

```diff
@@ -176,6 +103,16 @@ public void AddPeer(IPEndPoint endPoint, IPAddress source)
             this.peers.TryAdd(peerToAdd.Endpoint, peerToAdd);
         }
 
+        public void RemovePeer(IPEndPoint endPoint, IPAddress source)
+        {
+            if (!endPoint.Address.IsRoutable(true))
```

Comment:
Why do we have this condition?

Reply:
If endpoint was not Routable it has not been added to the peers list, so no need to go further right ?

Reply:
ok, I don't think it is needed, but ok. For me this is something that one has to think about when changing add peer method and does not really solve any problem, so I would not use it